doc/SONiC_QoS_Mgmt_HLD-1.1.md

```diff
@@ -0,0 +1,649 @@
+
+
+- Threshold Breach  
+
+  Augment Open Config YANG model with new definitions of Thresholds breach get.
+
+
+
+### 3.2.2 CLI
+
+#### 3.2.2.1 Configuration Commands
+
+- Configuring WRED
+````
+  sonic(config)# wred <name>
+  sonic(conf-wred-<name>)# random-detect color [green|yellow|red] minimum-threshold 100 maximum-threshold 300
+  drop-probability 40
+  sonic(conf-wred-<name>)# random-detect ecn <ecn_none/green/all>
+
+  sonic(config)# no wred <name>
+````
+- Configure Scheduler Policy for queues
+````
+  sonic(config)# scheduler-policy <name>
+  sonic(conf-sched-policy)# scheduler queue <q#>
+  sonic(conf-sched-policy-q)# priority
+  sonic(conf-sched-policy-q)# min <val>
+  sonic(conf-sched-policy-q)# min-burst <val>
+  sonic(conf-sched-policy-q)# max <val>
+  sonic(conf-sched-policy-q)# max-burst <val>
+  sonic(conf-sched-policy-q)# weight <val>
+
+  sonic(conf-sched-policy)# no scheduler <q#>
+  sonic(config)# no scheduler-policy <name>   
+````
+- Configure Scheduler Policy for port shaper
+
+````
+  sonic(config)# scheduler-policy <name>
+  sonic(conf-sched-policy)# scheduler port
+  sonic(conf-sched-policy-port)# max <val>
+  sonic(conf-sched-policy-port)# max-burst <val>
+````
+
+- Config DSCP to Traffic Class map  
+````
+  sonic(config)# qos-map dscp-to-tc-map <name>
+  sonic(conf-qos-map)# dscp {<dscp-value>} traffic-class <0..7>  
+  sonic(conf-qos-map)# no dscp {<dscp-value>}
+
+  sonic(config)# no dscp-to-tc-map <name>
+````
+- Config DOT1P to Traffic Class map  
+````
+  sonic(config)# qos-map dot1p-to-tc-map <name>
+  sonic(conf-qos-map)# dot1p {<dot1p-value>} traffic-class <0..7>  
+  sonic(conf-qos-map)# no dot1p {<dscp-value>}
+
+  sonic(config)# no dot1p-to-tc-map <name>
+````
+
+- Config Traffic Class to Queue map
+````
+  sonic(config)# qos-map tc-to-queue-map <name>
+  sonic(conf-qos-map)# traffic-class {<tc-value>} queue <0..7>
+````
+
+- Config interface QoS
+````
+  sonic(config)# qos interface <name>
+  sonic(conf-if-name)# dscp-to-tc-map <name>
+  sonic(conf-if-name)# dot1p-to-tc-map <name>
+  sonic(conf-if-name)# tc-to-queue-map <name>
+  sonic(conf-if-name)# pfc-priority <0-7>
+  sonic(conf-if-name)# pfc asymmetric
+  sonic(conf-if-name)# scheduler-policy <name>
+  sonic(conf-if-name)# queue <q#> wred-profile <wred-profile-name>
+  sonic(conf-if-name)# queue <q#> [ucast|mcast] threshold <val>
+
+  sonic(conf-if-name)# no dscp-to-tc-map <name>
+  sonic(conf-if-name)# no dot1p-to-tc-map <name>
+  sonic(conf-if-name)# no tc-to-queue-map <name>
+  sonic(conf-if-name)# no scheduler-policy <name>
+  sonic(conf-if-name)# no queue <q#> wred-profile <wred-profile-name>
+````
+
+#### 3.2.2.2 Show Commands
+
+- Show WRED
+````
+  >show wred {<name>}
+  Sample Output:
+    ---------------------------------------------------
+    Profile                : test
+    ecn                    : ecn_all
+    green-min-threshold    : 1048576

```

Comment:
Display units for the threshold attributes
`    green-min-threshold    : 1048576 Bytes
`

Reply:
Added KBytes